platforms/software/build-init/src/main/java/org/gradle/buildinit/plugins/internal/BuildScriptBuilderFactory.java

```diff
@@ -20,18 +20,20 @@
 import org.gradle.buildinit.InsecureProtocolOption;
 import org.gradle.buildinit.plugins.internal.modifiers.BuildInitDsl;
 
+import javax.annotation.Nullable;
+
 public class BuildScriptBuilderFactory {
     private final DocumentationRegistry documentationRegistry;
 
     public BuildScriptBuilderFactory(DocumentationRegistry documentationRegistry) {
         this.documentationRegistry = documentationRegistry;
     }
 
-    public BuildScriptBuilder scriptForNewProjects(BuildInitDsl dsl, String fileNameWithoutExtension, boolean useIncubatingAPIs) {
-        return new BuildScriptBuilder(dsl, documentationRegistry, fileNameWithoutExtension, useIncubatingAPIs, InsecureProtocolOption.FAIL);
+    public BuildScriptBuilder scriptForNewProjects(BuildInitDsl dsl, VersionCatalogDependencyRegistry versionCatalogDependencyRegistry, String fileNameWithoutExtension, boolean useIncubatingAPIs) {
```

Comment:
💅 This would be easier to read if there was an overload without the version catalog dependency registry parameter and this one with it non-nullable. Same for the `BuildScriptBuilder` constructor.

Reply:
There are now separate factory methods to create `BuildScriptBuilder` without version catalog usage